### PR TITLE
fix: handle hairpining timeout properly

### PR DESCRIPTION
**Pull request by dignifiedquire**

### CODE REVIEW

**Review by flub — APPROVED** [review]

**Comment by dignifiedquire** [comment]
@flub pushed a small additional simplification

File: src/hp/netcheck.rs [hunk 1]
```diff
@@ -793,21 +798,20 @@ impl ReportState {
...
-
-        tokio::select! {
-            biased;
-            _ = self.got_hair_stun.recv() => {
-                Some(true)
-            }
-            _ = hair_timeout => {
-                debug!("hair_check timeout");
-                Some(false)
+        match self.hair_timeout {
```
Comment (flub) [inline src/hp/netcheck.rs:801]: doesn't clippy want this as an if-let?  I don't mind though

**Review by flub — APPROVED** [review]

File: src/hp/netcheck.rs (hunk 1 above)
Comment (dignifiedquire) [reply src/hp/netcheck.rs:801]: it seems happy 😅

**Review by Arqu — APPROVED** [review]
Seems to have fixed the issue 👍

**Pull request merged**